app/models/user.rb

```diff
@@ -631,6 +639,6 @@ def votes_for_others
       .includes(comment: :user, story: :user)
       .where("(votes.comment_id is not null and comments.user_id <> votes.user_id) OR " \
```

Comment:
`.or()`

Reply:
One way to do this will be 
```
+    votes
+    .left_outer_joins(:story, :comment)
+    .includes(comment: :user, story: :user)
+    .where.not(votes: { comment_id: nil })
+    .where.not("comments.user_id = votes.user_id")
+    .or(
+      votes.where(comment_id: nil)
+           .where("stories.user_id <> votes.user_id")
+    )
+    .order(id: :desc) 
```
But this will change the underlying sql. This will require more wrestling with SQL to get it right.

Reply:
Could you paste both sql queries? I can't guess what the difference is?